rubi/rubi/client.py

```diff
@@ -0,0 +1,635 @@
+    ) -> None:
+        """Update the allowance for the base and quote assets of a pair if the current allowance is different from the
+        new allowance. This method also updates the Pair data structure so that the allowance can be read without having
+        to do a call to the chain.
+
+        :param pair_name: Name of the pair.
+        :type pair_name: str
+        :param new_base_asset_allowance: New allowance for the base asset. (optional, default is None).
+        :type new_base_asset_allowance: Optional[Decimal]
+        :param new_quote_asset_allowance: New allowance for the quote asset. (optional, default is None).
+        :type new_quote_asset_allowance: Optional[Decimal]
+        :raises PairDoesNotExistException: If the pair does not exist in the clients internal _pairs dict.
+        """
+        pair = self.get_pair(pair_name=pair_name)
+
+        if new_base_asset_allowance and pair.current_base_asset_allowance != new_base_asset_allowance:
+            self._update_asset_allowance(
+                asset=pair.base_asset,
+                spender=self.market.address,
+                new_allowance=new_base_asset_allowance
+            )
+            pair.update_base_asset_allowance(new_base_asset_allowance=new_base_asset_allowance)
+
+        if new_quote_asset_allowance and pair.current_base_asset_allowance != new_quote_asset_allowance:
+            self._update_asset_allowance(
+                asset=pair.quote_asset,
+                spender=self.market.address,
+                new_allowance=new_quote_asset_allowance
+            )
+            pair.update_quote_asset_allowance(new_quote_asset_allowance=new_quote_asset_allowance)
+
+    def get_pair(self, pair_name: str) -> Pair:
+        """Retrieves the Pair object associated with the specified pair name. If the pair does not exist
+        in the client, it raises a PairDoesNotExistException.
+
+        :param pair_name: Name of the pair.
+        :type pair_name: str
+        :return: The Pair object.
+        :rtype: Pair
+        :raises PairDoesNotExistException: If the pair does not exist in the client.
+        """
+        pair = self._pairs.get(pair_name)
+
+        if pair is None:
+            raise PairDoesNotExistException(
+                "add pair to the client using the add_pair method before placing orders for the pair"
+            )
+
+        return pair
+
+    def remove_pair(self, pair_name: str) -> None:
+        """Removes a pair from the client. It updates the pair's asset allowances
+        to zero and deletes the pair and its corresponding order book from the client.
+
+        :param pair_name: Name of the pair to remove.
+        :type pair_name: str
+        :raises PairDoesNotExistException: If the pair does not exist in the client before removal.
+        """
+        self.update_pair_allowance(
+            pair_name=pair_name,
+            new_base_asset_allowance=Decimal("0"),
```

Comment:
really like that we are keeping this set to zero, feels like best practice for end user.